syntax = "proto3";

// Model 消息,框架内部消息
package Model;



// Actor默认请求
// ResponseType ActorResponse
message ActorRequest	// IActorRequest
{
	int32 RpcId = 90;
}

// Actor默认回复消息
message ActorResponse	// IActorResponse
{
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
}





// =================== Location app ===================
// Location 添加对象消息
// ResponseType L2A_ObjectAddResponse
message A2L_ObjectAddRequest	// IActorRequest
{
	int32 RpcId = 90;
	
	int64 Key	= 1;
	int64 InstanceId = 2;
}

message L2A_ObjectAddResponse	// IActorResponse
{
	
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
}

// Location 锁住对象消息
// ResponseType L2A_ObjectLockResponse
message A2L_ObjectLockRequest	// IActorRequest
{
	int32 RpcId = 90;
	
	int64 Key	= 1;
}

message L2A_ObjectLockResponse	// IActorResponse
{
	
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
}

// Location 解锁对象消息
// ResponseType L2A_ObjectUnLockResponse
message A2L_ObjectUnLockRequest	// IActorRequest
{
	int32 RpcId = 90;
	
	int64 Key	= 1;
	// 原始的实例id
	int64 OldInstanceId = 2;
	// 现在的实例id
	int64 InstanceId = 3;
}

message L2A_ObjectUnLockResponse	// IActorResponse
{
	
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
}

// Location 移除对象消息
// ResponseType L2A_ObjectRemoveResponse
message A2L_ObjectRemoveRequest	// IActorRequest
{
	int32 RpcId = 90;
	
	int64 Key	= 1;
}

message L2A_ObjectRemoveResponse	// IActorResponse
{
	
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
}

// Location 获取对象消息
// ResponseType L2A_ObjectGetResponse
message A2L_ObjectGetRequest	// IActorRequest
{
	int32 RpcId = 90;
	
	int64 Key	= 1;
}

message L2A_ObjectGetResponse	// IActorResponse
{
	
	int32 RpcId = 90;
	int32 Error = 91;
	string Message = 92;	
	
	int64 InstanceId	= 1;
}





